src/programs.rkt

```diff
@@ -54,7 +53,7 @@
    [(real? x) (bf x)]
    [(bigfloat? x) x]
    [(complex? x)
-    (if (= (imag-part x) 0) (->bf (real-part x)) +nan.bf)]
+    (cons (->bf (real-part x)) (->bf (imag-part x)))]
```

Comment:
This is one of the relatively rare cases where it might be better to use a struct instead of just `cons` cells to store two numbers. Since bigcomplex numbers will show up a lot, distinguishing them easily from just pairs of bigfloats will be useful, and a struct lets us do that.